osdep/timer.c

```diff
@@ -103,8 +103,6 @@ struct timespec mp_time_us_to_realtime(int64_t time_us)
         diff_secs += 1;
         diff_nsecs -= 1000000000UL;
     }
-    // OSX can't deal with large timeouts. Also handles tv_sec/time_t overflows.
-    diff_secs = MPMIN(diff_secs, 10000000);
```

Comment:
macOS is apparently such a garbage platform that we still need this. How about just guarding it with `#if HAVE_DARWIN` instead.
https://github.com/dicej/ipmpsc/issues/4

Reply:
the hate isn't really necessary like this, or do we want to start another gnome situation?

Reply:
@Dudemanguy: In fact, I reintroduced this later, set to 1000 days... to guard for time_t overflow. We call wait function with infinite time in few places, so this has to be clamped anyway. I hope this will be enough also for OSX.

@Akemi: Don't worry winpthreads is even more garbage it cannot handle wait times that are close less than current time, but also not more than 1s...

Reply:
Everything sucks. Well I guess we can just try it and see what happens.

Reply:
I should probably rebase and remove this commit. Was too lazy to do that.

Reply:
Done now...

Reply:
We really should replace things like `wait_wakeup(ctx, INT64_MAX);` to call `pthread_cond_wait` instead. But it is hard to track all of them, when the wait time comes from variable.